README.txt
2013-06-07 Wasta Backup Readme.


debian/changelog
wasta-backup (1.3.7) noble; urgency=medium

  * allow building for 24.04

 -- Justin Luth <jluth@mail.com>  Thu, 16 Nov 2023 14:21:31 -0500

wasta-backup (1.3.6) bionic; urgency=medium

  * converted logo text to path so scales better

 -- Rik Shaw <rikshaw76@gmail.com>  Tue, 24 Mar 2020 16:19:59 +0300

wasta-backup (1.3.5) bionic; urgency=medium

  * padded logo

 -- Rik Shaw <rikshaw76@gmail.com>  Thu, 19 Mar 2020 16:00:54 +0300

wasta-backup (1.3.4) bionic; urgency=medium

  * fix another missing quotes on config file path

 -- Justin Luth <jluth@mail.com>  Tue, 17 Mar 2020 09:26:59 +0300

wasta-backup (1.3.3) bionic; urgency=medium

  * fix missing quotes on backup path

 -- Justin Luth <jluth@mail.com>  Mon, 16 Mar 2020 07:29:27 +0300

wasta-backup (1.3.2) bionic; urgency=medium

  * ensure lastUsedConfigs really are the most recent
  * hide "USB config" if no USB inserted/configured

 -- Justin Luth <jluth@mail.com>  Thu, 30 Jan 2020 21:21:08 +0300

wasta-backup (1.3.1) bionic; urgency=medium

  * icon name fix

 -- Rik Shaw <rikshaw76@gmail.com>  Sat, 25 Jan 2020 16:06:19 +0300

wasta-backup (1.3.0) bionic; urgency=medium

  * rework 'About' and add 'Configuration Guide'

 -- Rik Shaw <rikshaw76@gmail.com>  Sat, 25 Jan 2020 15:00:17 +0300

wasta-backup (1.2.8) focal; urgency=medium

  * Add Paratext9Projects to suggested/default backup list
  * Allow more than 20 backup items in backupDirs.txt

 -- Justin Luth <jluth@mail.com>  Fri, 24 Jan 2020 13:43:12 +0300

wasta-backup (1.2.7) bionic; urgency=medium

  * French l10n updated
  * mainwindow.cpp: adding ":" to translation string

 -- Rik Shaw <rikshaw76@gmail.com>  Wed, 16 Jan 2019 16:06:23 +0300

wasta-backup (1.2.6) bionic; urgency=medium

  * mainwindow.cpp: cleaning up tr() strings to make more simple
    to translate
  - Using xdg-user-dir to retrieve user's DESKTOP and DOCUMENTS locations
    to account for locale differences in folder names.

 -- Rik Shaw <rikshaw76@gmail.com>  Sat, 01 Dec 2018 20:44:55 +0300

wasta-backup (1.2.5) bionic; urgency=medium

  * mainwindow.cpp: restoreDeleted: if folder of deleted item to
    restore is a symlink, follow to REAL path
    (restorePrevious did not have this issue)

 -- Rik Shaw <rikshaw76@gmail.com>  Wed, 21 Nov 2018 12:58:21 +0300

wasta-backup (1.2.4) bionic; urgency=medium

  * disabling postinst pending feedback from rdiff-backup developers

 -- Rik Shaw <rikshaw76@gmail.com>  Wed, 21 Nov 2018 11:09:39 +0300

wasta-backup (1.2.3) bionic; urgency=medium

  * postinst: correcting sed to comment out line that crashes with exFAT

 -- Rik Shaw <rikshaw76@gmail.com>  Tue, 20 Nov 2018 17:32:09 +0300

wasta-backup (1.2.2) bionic; urgency=medium

  * lintian tag cleanup

 -- Rik Shaw <rikshaw76@gmail.com>  Fri, 16 Nov 2018 20:02:40 +0300

wasta-backup (1.2.1) bionic; urgency=medium

  * mainwindow.cpp: correct QFileInfo to QFile for exist check

 -- Rik Shaw <rikshaw76@gmail.com>  Fri, 16 Nov 2018 17:22:19 +0300

wasta-backup (1.2.0) bionic; urgency=medium

  * mainwindow.cpp: when backing up, if source is symlink will follow
      to REAL path
    - Attempt restore if either of the following true:
      1. restoreTime == now AND is in the current backup
      2. restoreTime <> now and found with rdiff-backup --list-at-time 
  * wasta-backup-postinst.sh: added to tweak rdiff-backup for
      exFAT filesystem compatibility

 -- Rik Shaw <rikshaw76@gmail.com>  Fri, 16 Nov 2018 14:51:32 +0300

wasta-backup (1.1.4) xenial; urgency=medium

  * mainwindow.cpp: adding Paratext8Projects to default folders

 -- Rik Shaw <rikshaw76@gmail.com>  Wed, 23 May 2018 08:53:19 +0300

wasta-backup (1.1.3) xenial; urgency=medium

  * mainwindow.cpp: replacing ' with \" for shell commands to accomodate
    single quotes in folder/file names.
  * adding fr l10n

 -- Rik Shaw <rikshaw76@gmail.com>  Fri, 31 Mar 2017 13:06:57 +0300

wasta-backup (1.1.2) xenial; urgency=medium

  * correcting l10n location

 -- Rik Shaw <rikshaw76@gmail.com>  Fri, 30 Dec 2016 15:43:59 +0300

wasta-backup (1.1.1) xenial; urgency=medium

  * placing l10n files in /usr/share/wasta-backup
  * cleaning up debian folder

 -- Rik Shaw <rikshaw76@gmail.com>  Fri, 30 Dec 2016 13:44:21 +0300

wasta-backup (1.1.0) xenial; urgency=medium

  * refactored code and interface to support translation strings
  * added l10n/* folder and files for translations
  * made sure rdiff-backup commands use "2>&1" to capture warning messages 

 -- Rik Shaw <rikshaw76@gmail.com>  Fri, 30 Dec 2016 12:14:00 +0300

wasta-backup (1.0.11) xenial; urgency=medium

  * reverting accidental commit of main.cpp that was made
    attempting to provide localization for wasta-backup.

 -- Rik Shaw <rikshaw76@gmail.com>  Thu, 05 May 2016 18:19:09 +0300

wasta-backup (1.0.10) xenial; urgency=medium

  * jcl: only autolaunch on USB drive insertion, not all block devices

    Previously the udev rule launched on any SUBSYSTEM that was       
    considered a block device, including normal drive partitions.
    This could be triggered by creating a /media/wasta-offline folder
    for example.

    USB seems like the only kind of device insertion that we are      
    interested in auto-launching wasta-offline.

 -- Rik Shaw <rikshaw76@gmail.com>  Thu, 05 May 2016 17:39:09 +0300

wasta-backup (1.0.9) wily; urgency=medium

  * wasta-backup-udev: account for machines that have "broken at" due to
    remastersys throwing away the config file /var/spool/cron/atjobs/.SEQ
    Without that file, at doesn't work and autolaunch is broken. Fixed
    by running "sudo dpkg-reconfigure at", but patching here so if that
    file isn't found, we revert to the old method of launching wasta-backup.
    WARNING: if using 15.10+ and that file doesn't exist, the old method
    is used, then automount of ALL USBs is broken.  So we need to ensure
    that wasta-remastersys properly configs "at" in the firstboot script.

 -- Rik Shaw <rikshaw76@gmail.com>  Thu, 12 Nov 2015 16:39:16 +0300


moc_mainwindow.cpp
/****************************************************************************
** Meta object code from reading C++ file 'mainwindow.h'
**
** Created by: The Qt Meta Object Compiler version 63 (Qt 4.8.7)
**
** WARNING! All changes made in this file will be lost!
*****************************************************************************/

#include "mainwindow.h"
#if !defined(Q_MOC_OUTPUT_REVISION)
#error "The header file 'mainwindow.h' doesn't include <QObject>."
#elif Q_MOC_OUTPUT_REVISION != 63
#error "This file was generated using the moc from 4.8.7. It"
#error "cannot be used with the include files from this version of Qt."
#error "(The moc has changed too much.)"
#endif

QT_BEGIN_MOC_NAMESPACE
static const uint qt_meta_data_MainWindow[] = {

 // content:
       6,       // revision
       0,       // classname
       0,    0, // classinfo
      28,   14, // methods
       0,    0, // properties
       0,    0, // enums/sets
       0,    0, // constructors
       0,       // flags
       0,       // signalCount

 // slots: signature, parameters, type, tag, flags
      12,   11,   11,   11, 0x08,
      38,   11,   11,   11, 0x08,
      93,   72,   64,   11, 0x08,
     121,  116,   11,   11, 0x08,
     139,   11,   11,   11, 0x08,
     157,   11,   11,   11, 0x08,
     187,   11,   11,   11, 0x08,
     216,   11,   11,   11, 0x08,
     245,   11,   11,   11, 0x08,
     272,   11,   11,   11, 0x08,
     304,   11,   11,   11, 0x08,
     338,   11,   11,   11, 0x08,
     365,   11,   11,   11, 0x08,
     400,  394,   11,   11, 0x08,
     443,   11,   11,   11, 0x08,
     478,   11,   11,   11, 0x08,
     505,   11,   11,   11, 0x08,
     537,   11,   11,   11, 0x08,
     570,   11,   11,   11, 0x08,
     586,   11,   11,   11, 0x08,
     628,  623,   11,   11, 0x08,
     696,  665,   11,   11, 0x08,
     739,   11,   11,   11, 0x08,
     778,   11,   11,   11, 0x08,
     805,   11,   11,   11, 0x08,
     853,  845,  840,   11, 0x08,
     881,  872,   11,   11, 0x08,
     919,  906,   11,   11, 0x08,

       0        // eod
};

static const char qt_meta_stringdata_MainWindow[] = {
    "MainWindow\0\0setPreferredDestination()\0"
    "on_backupButton_clicked()\0QString\0"
    "command,giveFeedback\0shellRun(QString,bool)\0"
    "data\0writeLog(QString)\0loadConfigFiles()\0"
    "on_restorePrevRadio_clicked()\0"
    "on_restoreDelRadio_clicked()\0"
    "on_restoreAllRadio_clicked()\0"
    "on_restoreButton_clicked()\0"
    "on_changeDeviceButton_clicked()\0"
    "on_selectPrevItemButton_clicked()\0"
    "on_prevFileRadio_clicked()\0"
    "on_prevFolderRadio_clicked()\0index\0"
    "on_backupRestoreWidget_currentChanged(int)\0"
    "on_selectDelFolderButton_clicked()\0"
    "on_viewLogButton_clicked()\0"
    "on_cancelBackupButton_clicked()\0"
    "on_cancelRestoreButton_clicked()\0"
    "cancelProcess()\0on_openRestoreFolderButton_clicked()\0"
    "arg1\0on_restoreAllCheck_stateChanged(int)\0"
    "renameItem,targetItem,restUser\0"
    "renameRestoreItem(QString,QString,QString)\0"
    "on_actionBackupOnlyImportant_changed()\0"
    "on_actionAbout_triggered()\0"
    "on_undoLastRestoreButton_clicked()\0"
    "bool\0dirName\0removeDir(QString)\0"
    "inputDir\0setTargetDevice(QString)\0"
    "machineValue\0on_machineCombo_currentIndexChanged(QString)\0"
};

void MainWindow::qt_static_metacall(QObject *_o, QMetaObject::Call _c, int _id, void **_a)
{
    if (_c == QMetaObject::InvokeMetaMethod) {
        Q_ASSERT(staticMetaObject.cast(_o));
        MainWindow *_t = static_cast<MainWindow *>(_o);
        switch (_id) {
        case 0: _t->setPreferredDestination(); break;
        case 1: _t->on_backupButton_clicked(); break;
        case 2: { QString _r = _t->shellRun((*reinterpret_cast< QString(*)>(_a[1])),(*reinterpret_cast< bool(*)>(_a[2])));
            if (_a[0]) *reinterpret_cast< QString*>(_a[0]) = _r; }  break;
        case 3: _t->writeLog((*reinterpret_cast< QString(*)>(_a[1]))); break;
        case 4: _t->loadConfigFiles(); break;
        case 5: _t->on_restorePrevRadio_clicked(); break;
        case 6: _t->on_restoreDelRadio_clicked(); break;
        case 7: _t->on_restoreAllRadio_clicked(); break;
        case 8: _t->on_restoreButton_clicked(); break;
        case 9: _t->on_changeDeviceButton_clicked(); break;
        case 10: _t->on_selectPrevItemButton_clicked(); break;
        case 11: _t->on_prevFileRadio_clicked(); break;
        case 12: _t->on_prevFolderRadio_clicked(); break;
        case 13: _t->on_backupRestoreWidget_currentChanged((*reinterpret_cast< int(*)>(_a[1]))); break;
        case 14: _t->on_selectDelFolderButton_clicked(); break;
        case 15: _t->on_viewLogButton_clicked(); break;
        case 16: _t->on_cancelBackupButton_clicked(); break;
        case 17: _t->on_cancelRestoreButton_clicked(); break;
        case 18: _t->cancelProcess(); break;
        case 19: _t->on_openRestoreFolderButton_clicked(); break;
        case 20: _t->on_restoreAllCheck_stateChanged((*reinterpret_cast< int(*)>(_a[1]))); break;
        case 21: _t->renameRestoreItem((*reinterpret_cast< QString(*)>(_a[1])),(*reinterpret_cast< QString(*)>(_a[2])),(*reinterpret_cast< QString(*)>(_a[3]))); break;
        case 22: _t->on_actionBackupOnlyImportant_changed(); break;
        case 23: _t->on_actionAbout_triggered(); break;
        case 24: _t->on_undoLastRestoreButton_clicked(); break;
        case 25: { bool _r = _t->removeDir((*reinterpret_cast< const QString(*)>(_a[1])));
            if (_a[0]) *reinterpret_cast< bool*>(_a[0]) = _r; }  break;
        case 26: _t->setTargetDevice((*reinterpret_cast< QString(*)>(_a[1]))); break;
        case 27: _t->on_machineCombo_currentIndexChanged((*reinterpret_cast< const QString(*)>(_a[1]))); break;
        default: ;
        }
    }
}

const QMetaObjectExtraData MainWindow::staticMetaObjectExtraData = {
    0,  qt_static_metacall 
};

const QMetaObject MainWindow::staticMetaObject = {
    { &QMainWindow::staticMetaObject, qt_meta_stringdata_MainWindow,
      qt_meta_data_MainWindow, &staticMetaObjectExtraData }
};

#ifdef Q_NO_DATA_RELOCATION
const QMetaObject &MainWindow::getStaticMetaObject() { return staticMetaObject; }
#endif //Q_NO_DATA_RELOCATION

const QMetaObject *MainWindow::metaObject() const
{
    return QObject::d_ptr->metaObject ? QObject::d_ptr->metaObject : &staticMetaObject;
}

void *MainWindow::qt_metacast(const char *_clname)
{
    if (!_clname) return 0;
    if (!strcmp(_clname, qt_meta_stringdata_MainWindow))
        return static_cast<void*>(const_cast< MainWindow*>(this));
    return QMainWindow::qt_metacast(_clname);
}

int MainWindow::qt_metacall(QMetaObject::Call _c, int _id, void **_a)
{
    _id = QMainWindow::qt_metacall(_c, _id, _a);
    if (_id < 0)
        return _id;
    if (_c == QMetaObject::InvokeMetaMethod) {
        if (_id < 28)
            qt_static_metacall(this, _c, _id, _a);
        _id -= 28;
    }
    return _id;
}
QT_END_MOC_NAMESPACE
